airbyte-integrations/bases/base-python/base_python/cdk/abstract_source.py

```diff
@@ -78,15 +79,25 @@ def read(
         # TODO assert all streams exist in the connector
         # get the streams once in case the connector needs to make any queries to generate them
         stream_instances = {s.name: s for s in self.streams(config)}
-        for configured_stream in catalog.streams:
+        with EventTimer(self.name) as timer:
             try:
-                stream_instance = stream_instances[configured_stream.stream.name]
-                yield from self._read_stream(
-                    logger=logger, stream_instance=stream_instance, configured_stream=configured_stream, connector_state=connector_state
-                )
-            except Exception as e:
-                logger.exception(f"Encountered an exception while reading stream {self.name}")
-                raise e
+                for configured_stream in catalog.streams:
+                    try:
+                        stream_instance = stream_instances[configured_stream.stream.name]
+                        timer.start_event(configured_stream.stream.name)
```

Comment:
What is exception while _read_stream occure? Why you dont using context manager here as well?

Reply:
fixed